src/cloud-api-adaptor/pkg/podnetwork/workernode.go

```diff
@@ -186,6 +189,26 @@ func (n *workerNode) Inspect(nsPath string) (*tunneler.Config, error) {
 		config.Routes = append(config.Routes, r)
 	}
 
+	// Add route for the subnet CIDRs in the new namespace
+	if n.PodSubnetCIDRs != nil {
+		for _, cidr := range n.PodSubnetCIDRs {
+			prefix, err := netip.ParsePrefix(cidr)
+			if err != nil {
+				logger.Printf("failed to parse CIDR %q: %s", cidr, err)
+				continue
+			}
+			route := &tunneler.Route{
+				Dst: prefix,
+				// The gateway address is 0.0.0.0
+				GW:       netip.Addr{},
+				Dev:      podInterface,
+				Scope:    netops.RouteScope(netlink.SCOPE_LINK),
+				Protocol: netops.RouteProtocol(unix.RTPROT_KERNEL),
+			}
```

Comment:
GW should be the actual gateway address that is specified in the default route.

The `findPrimaryInterface` function that is called at Line 129 internally identifies the default route. I think we can modify the function to return the gateway address of the default route.
```
	podInterface, gatewayAddr, err := findPrimaryInterface(podNS)
```
```
			route := &tunneler.Route{
				Dst: prefix,
				GW: gatewayAddr,
				Dev: podInterface,
			}
```